pkg/dockercreds/cred_helper_test.go

```diff
@@ -0,0 +1,131 @@
+package dockercreds_test
+
+import (
+	"os"
+	"strconv"
+	"testing"
+
+	"github.com/sclevine/spec"
+	"github.com/stretchr/testify/require"
+
+	"github.com/vmware-tanzu/kpack-cli/pkg/dockercreds"
+)
+
+func TestCredHelper(t *testing.T) {
+	spec.Run(t, "TestCredHelper", testCredHelper)
+}
+
+const (
+	envVarRegistryUrl      = "KP_REGISTRY_HOSTNAME_TEST_VAR"
+	envVarRegistryUser     = "KP_REGISTRY_USERNAME_TEST_VAR"
+	envVarRegistryPassword = "KP_REGISTRY_PASSWORD_TEST_VAR"
+)
+
+type registryVariable struct {
+	URLName       string
+	URLValue      string
+	UserName      string
+	UserValue     string
+	PasswordName  string
+	PasswordValue string
+}
+
+func getValuesUnderTest() []registryVariable {
+	values := make([]registryVariable, 0, 1+len(digitToWord()))
+	values = append(values, registryVariable{
+		URLName:       envVarRegistryUrl,
+		URLValue:      "some-registry.io",
+		UserName:      envVarRegistryUser,
+		UserValue:     "some-registry-user",
+		PasswordName:  envVarRegistryPassword,
+		PasswordValue: "some-registry-password",
+	})
+	for digit, word := range digitToWord() {
+		values = append(values, registryVariable{
+			URLName:       envVarRegistryUrl + "_" + strconv.Itoa(digit),
+			URLValue:      word + "-registry.io",
+			UserName:      envVarRegistryUser + "_" + strconv.Itoa(digit),
+			UserValue:     word + "-registry-user",
+			PasswordName:  envVarRegistryPassword + "_" + strconv.Itoa(digit),
+			PasswordValue: word + "-registry-password",
+		})
+	}
+	return values
+}
+
+func digitToWord() map[int]string {
+	return map[int]string{
+		0:   "zero",
+		1:   "one",
+		2:   "two",
+		3:   "three",
+		9:   "nine",
+		10:  "ten",
+		111: "one-hundred-eleven",
+	}
+}
+
+func testCredHelper(t *testing.T, when spec.G, it spec.S) {
+
+	when("one registry credential is provided by environment variables and Get is called", func() {
+		var credHelper *dockercreds.CredHelper
+
+		it.Before(func() {
+			require.NoError(t, os.Setenv(envVarRegistryUrl, "foo-registry.io"))
+			require.NoError(t, os.Setenv(envVarRegistryUser, "foo-registry-user"))
+			require.NoError(t, os.Setenv(envVarRegistryPassword, "foo-registry-password"))
+		})
+
+		it.After(func() {
+			require.NoError(t, os.Unsetenv(envVarRegistryUrl))
+			require.NoError(t, os.Unsetenv(envVarRegistryUser))
+			require.NoError(t, os.Unsetenv(envVarRegistryPassword))
+		})
```

Comment:
You don't need to worry about cleaning up env vars since child processes cannot interact with the parent's env vars (other than inheriting a copy).

For example, `go run`-ing this program won't unset `$HOME` nor add `$FOOBAR` to the shell
```go
package main
import "os"
func main() {
	os.Setenv("FOOBAR", "BAZ")
	os.Unsetenv("HOME")
}
```